src/config/types.ts

```diff
@@ -23,8 +23,9 @@ export const PermissiveBoolean = new t.Type<boolean, t.TypeOf<PermissiveType>>(
 				case 'string':
 				case 'boolean':
 				case 'number':
-					const val = checkTruthy(v);
-					if (val == null) {
+					const strict = false; // allow undefined
+					const val = checkBoolean(v, strict);
```

Comment:
The reason this code references checkBoolean instead of checkTruthy like all other places is because considering the effects of making values return only true or false means that if an invalid value is passed it will might turn settings off rather then throw an error like before. So this function exists to improve checkTruthy but still follow the previous logic.